backend/controllers/routes.js

```diff
@@ -1,18 +1,127 @@
 // Boilerplate https://github.com/Intro-to-SE-Spring-2020/Chirpr/blob/master/backend/controllers/auth.js
 
-// import models
+// import
 const User = require('../models/user')
 const Route = require('../models/route');
+const {v4 : uuidv4} = require('uuid')
 
-exports.createRoute = async (req, res) => {
+
+exports.getRoute = async (req, res) => {
+
+  const { routeid } = req.params
+
+  // prevents bypassing auth with another usename
+  if(!routeid)
+  {
+    res.status(400);
+    res.json({
+      status: '400',
+      success: 'false',
+      msg: 'Bad Request'
+    })
+    return res
+  }
 
   try {
+    // check if route exists
+    const route = await Route.findOne({ routeid: routeid})
+
+    if (!route) {
+      res.status(404);
       res.json({
-        msg: 'Middleware test successful'
+        status: '404',
+        success: 'false',
+        msg: 'Route does not exist'
       })
+      return res
+    }
+
+    var data = {
+      routeid: route.routeid,
+      routedescription: route.routedescription,
+      username: route.username,
+      routetitle: route.routetitle,
+      routetype: route.routetype,
+      routetime: route.routetime,
+      routedistance: route.routedistance,
+      photos: route.photos,
+      mapdata: route.mapdata
+    }
+
+    // return the data
+    res.status(200);
+    res.json({
+      status: '200',
+      success: 'true',
+      data: data,
+      msg: 'Route retrieved'
+    })
+    return res
+
+  } catch (err) {
+    console.error(err)
+    res.status(500);
+    res.json({
+      status: '500',
+      success: 'false',
+      msg: 'Internal Server error'
+    })
+    return res
+  }
+}
+
+
+exports.createRoute = async (req, res) => {
+
+  var { username, routetitle, routetype, routetime, routedescription, photos, mapdata, routedistance } = req.body
+
+  if(!username || !routetitle || !routetype || !routetime || !routedescription || !photos || !mapdata || !routedistance || typeof username != 'string' ||
+    typeof routetitle != 'string' || typeof routetype != 'string' ||typeof routetime != 'string' || typeof routedescription != 'string' || typeof photos != 'object'
+    ||typeof mapdata != 'object' ||typeof routedistance != 'number' || !mapdata['coordinates'] || !mapdata['type'] || typeof mapdata['coordinates'] != 'object' ||
+      typeof mapdata['type'] != 'string'){
+    res.status(400);
+    res.json({
+      status: '400',
+      success: 'false',
+      msg: 'Bad Request'
+    })
+    return res
+  }
+
```

Comment:
Can we confirm the structure of `mapdata`?

I think we set it to be a list of polylines? An example from the frontend is:
```
route: [
            [
                [51.5, -0.1],
                [51.5, -0.12],
                [51.52, -0.12],
            ],
            [
                [51.5, -0.05],
                [51.5, -0.06],
                [51.52, -0.06],
            ],
        ]
```

From the `if` conditions here I see we are using the old version of `mapdata`. It was prematurely set up for an earlier feature.

Reply:
After consulting with @Mitul264 , the structure of `mapdata` attribute in the `Route` model will simply be changed to `{}`. Closing the conversation.